x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/snapshotlifecycle/SnapshotLifecycleServiceTests.java

```diff
@@ -80,6 +81,48 @@ public void testRepositoryExistenceForMissingRepo() {
         assertThat(e.getMessage(), containsString("no such repository [repo]"));
     }
 
+    public void testNothingScheduledWhenNotRunning() {
+        ClockMock clock = new ClockMock();
+        try (ThreadPool threadPool = new TestThreadPool("test");
+             ClusterService clusterService = ClusterServiceUtils.createClusterService(threadPool);
+             SnapshotLifecycleService sls = new SnapshotLifecycleService(Settings.EMPTY,
+                 () -> new FakeSnapshotTask(e -> logger.info("triggered")), clusterService, clock)) {
+
+            sls.offMaster();
+
+            SnapshotLifecyclePolicyMetadata newPolicy = SnapshotLifecyclePolicyMetadata.builder()
+                .setPolicy(createPolicy("foo", "*/1 * * * * ?"))
+                .setHeaders(Collections.emptyMap())
+                .setVersion(2)
+                .setModifiedDate(2)
+                .build();
+            Map<String, SnapshotLifecyclePolicyMetadata> policies = new HashMap<>();
+            policies.put(newPolicy.getPolicy().getId(), newPolicy);
+            ClusterState emptyState = createState(new SnapshotLifecycleMetadata(Collections.emptyMap(), OperationMode.RUNNING));
+            ClusterState state = createState(new SnapshotLifecycleMetadata(policies, OperationMode.RUNNING));
+
+            sls.clusterChanged(new ClusterChangedEvent("1", state, emptyState));
+
+            // Since the service does not think it is master, it should not be triggered or scheduled
+            assertThat(sls.getScheduler().scheduledJobIds(), equalTo(Collections.emptySet()));
+
+            sls.onMaster();
```

Comment:
Do we want to make sure that jobs actually got scheduled here before going on to stop SLM and making sure they get un-scheduled?

Reply:
Sure, I pushed a commit to add that to the test